src/field/binary/mod.rs

```diff
@@ -0,0 +1,107 @@
+//! Contains implementation of binary field and extension fields
+use std::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign},
+};
+
+use crate::field::FiniteField;
+
+pub mod extension;
+#[cfg(test)] pub mod tests;
+
+/// binary field containing element `{0,1}`
+#[derive(Debug, Default, Hash, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BinaryField(u8);
```

Comment:
Perhaps a type alias instead with:
```suggestion
pub type BinaryField = PrimeField<2>;
```

Reply:
I believe that would collapse a lot of this module here.

Reply:
that'll work flawlessly, but i believe it wouldn't allow us to showcase the main properties of binary field, i.e. addition being bitwise XOR and multiplication bitwise AND.

Reply:
What if you use a trait impl override/default/specialization? There are some Rust unstable feats that may help here to let us specialize to the binary case.

I thought we had it in the project, but don't seem to anymore. Anyway, here is an RFC for it:
https://rust-lang.github.io/rfcs/1210-impl-specialization.html

There could be others.

Reply:
@Autoparallel i need some more time to check and iterate on this. so, i'll open a new PR for this. don't want to block AES implementations due to this change, so merging it.